source/Calamari.Common/Features/Scripting/Bash/Bootstrap.sh

```diff
@@ -263,4 +263,92 @@ function log_environment_information
 	echo "##octopus[stdout-default]"
 }
 
-log_environment_information
\ No newline at end of file
+log_environment_information
+
+function decrypt_and_parse_variables {
+    local encrypted="$1"
+    local iv="$2"
+
+    local decrypted
+    decrypted=$(decrypt_variable "$encrypted" "$iv")
+    declare -gA octopus_parameters=()
+
+    key_byte_lengths=()
+    concatenated_hex=""
+    while IFS= read -r hex_key; do
+        hex_key="${hex_key//$'\n'/}"
+        key_byte_len=$(( ${#hex_key} / 2 ))
+        key_byte_lengths+=("$key_byte_len")
+        concatenated_hex+="${hex_key}"
+    done <<< "$decrypted"
+    decoded_output=$(hex_to_string "$concatenated_hex")
+    
+    section_end=$(date +%s%3N)
+
+    section_start=$(date +%s%3N)
+    decoded_keys=()
+
+    tmp_file=$(mktemp)
+    printf "%s" "$decoded_output" > "$tmp_file"
+    exec 3<"$tmp_file"
+
+    for idx in "${!key_byte_lengths[@]}"; do
+        key_byte_len="${key_byte_lengths[idx]}"
+        read -r -N "$key_byte_len" decoded_key <&3
+        decoded_keys+=("$decoded_key")
+    done
+
+    exec 3<&-
+    rm -f "$tmp_file"
+
+    for key in "${decoded_keys[@]}"; do
+        # Value is empty for now, since we write the hex encoded key/values to disk we don't want to write the values
+        # https://github.com/OctopusDeploy/Calamari/pull/1441/files is an example implementation that includes the values
+        octopus_parameters["$key"]=""
+    done
+}
+
+hex_to_string() {
```

Comment:
This can be removed if we can depend on `xxd`. This is generally supported but not available everywhere, we could check if it is available and only add the octopus_parameters if it is.

Reply:
`hexdump` seems generally available on Nix, but is not installed by default on mac